lib/expression/set.go

```diff
@@ -18,65 +18,31 @@
 
 package expression
 
-import (
-	"golang.org/x/exp/maps"
-)
+import "github.com/gravitational/teleport/lib/utils"
 
-// Set is a map of type string key and struct values.
-type Set map[string]struct{}
-
-// NewSet returns Set from given string slice.
-func NewSet(values ...string) Set {
-	s := make(Set, len(values))
-	for _, value := range values {
-		s[value] = struct{}{}
-	}
-	return s
-}
-
-func (s Set) add(values ...string) Set {
-	out := s.clone()
-	for _, value := range values {
-		out[value] = struct{}{}
-	}
-	return out
-}
```

Comment:
The add and remove methods on the generic set mutate the receiver, while this one used to only mutate the copy. Unfortunately it looks like we had zero test coverage for that. I'm going to revert the behaviour on expression.Set to avoid mutating the receiver.

Reply:
I will revert because I think this could be a subtle breaking change to existing login rules and other expressions that customers have defined

Reply:
Nevermind, you did the clone inline in the parser spec definition, nothing to see here folks 😅